cardano-cli/src/Cardano/CLI/Run/Friendly.hs

```diff
@@ -154,8 +159,110 @@ friendlyUpdateProposal = \case
 
 friendlyCertificates :: TxCertificates ViewTx era -> Aeson.Value
 friendlyCertificates = \case
-  TxCertificatesNone -> Null
-  TxCertificates _ cs _ -> toJSON $ map textShow cs
+  TxCertificatesNone    -> Null
+  TxCertificates _ cs _ -> array $ map friendlyCertificate cs
+
+friendlyCertificate :: Certificate -> Aeson.Value
+friendlyCertificate =
+  object . (:[]) .
+  \case
+    -- Stake address certificates
+    StakeAddressRegistrationCertificate credential ->
+      "stake address registration" .=
+        object [friendlyStakeCredential "" credential]
+    StakeAddressDeregistrationCertificate credential ->
+      "stake address deregistration" .=
+        object [friendlyStakeCredential "" credential]
+    StakeAddressDelegationCertificate credential poolId ->
+      "stake address delegation" .=
+        object [friendlyStakeCredential "" credential, "pool" .= poolId]
+
+    -- Stake pool certificates
+    StakePoolRegistrationCertificate parameters ->
+      "stake pool registration" .= friendlyStakePoolParameters parameters
+    StakePoolRetirementCertificate poolId epochNo ->
+      "stake pool retirement" .= object ["pool" .= poolId, "epoch" .= epochNo]
+
+    -- Special certificates
+    GenesisKeyDelegationCertificate genesisKeyHash delegateKeyHash vrfKeyHash ->
+      "genesis key delegation" .=
+        object
+          [ "genesis key hash"  .= serialiseToRawBytesHexText genesisKeyHash
+          , "delegate key hash" .= serialiseToRawBytesHexText delegateKeyHash
+          , "VRF key hash"      .= serialiseToRawBytesHexText vrfKeyHash
+          ]
+    MIRCertificate pot target ->
+      "MIR" .= object ["pot" .= friendlyMirPot pot, friendlyMirTarget target]
+
+friendlyMirTarget :: MIRTarget -> (Text, Aeson.Value)
+friendlyMirTarget = \case
+  StakeAddressesMIR addresses ->
+    "target stake addresses" .=
+      [ object
+          [ friendlyStakeCredential "" credential
+          , "amount" .= friendlyLovelace lovelace
+          ]
+      | (credential, lovelace) <- addresses
+      ]
+  SendToReservesMIR amount -> "send to reserves" .= friendlyLovelace amount
+  SendToTreasuryMIR amount -> "send to treasury" .= friendlyLovelace amount
+
+friendlyStakeCredential :: Text -> StakeCredential -> (Text, Aeson.Value)
```

Comment:
I don't understand the point of the "subkey" especially as it's only used once.

Reply:
rewrote without it